ops/_private/timeconv.py

```diff
@@ -63,7 +63,7 @@ def parse_rfc3339(s: str) -> datetime.datetime:
                              microsecond=microsecond, tzinfo=tz)
 
 
-def parse_duration(s: str) -> datetime.timedelta:
+def parse_duration(s: str) -> datetime.timedelta:  # noqa: C901
```

Comment:
[C901](https://www.flake8rules.com/rules/C901.html) is "function is too complex", right? I think that's not appropriate for a linter and we should turn this off in Ruff's config. Whether or not a function is "too complex" is a nuanced question and should be judged in code review.

Reply:
Yeah, I tend to agree, and had this on my list of things to figure out. In the existing (main) config, we have:

```
select = ["E", "W", "F", "C", "N", "R", "D", "H"]
```

And we don't ignore anything starting with `C`. So it seems like this ought to be running at the moment, but I think that can't be the case, because I've bumped the maximum value up very high and it's still picking things up in the branch and nothing is detected in main.

Maybe the `C` in main is doing nothing because a plug-in isn't installed? Will dig more.

Reply:
It looks like the mccabe check is only done with flake8 if you pass a maximum complexity value, so we were previously only semi-enabling the check.